lib/util/include/hse_util/map.h

```diff
@@ -0,0 +1,90 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (C) 2022 Micron Technology, Inc.  All rights reserved.
+ */
+
+#include <hse_util/hse_err.h>
+
+#ifndef HSE_CORE_MAP_H
+#define HSE_CORE_MAP_H
+
+struct map;
+
+struct map *
+map_create(size_t initial_cnt);
+
+void
+map_destroy(struct map *map);
+
+merr_t
+map_insert(struct map *map, uint64_t key, uintptr_t value);
+
+static HSE_ALWAYS_INLINE merr_t
+map_insert_ptr(struct map *map, uint64_t key, void *val)
+{
+    return map_insert(map, key, (uintptr_t)val);
+}
+
+bool
+map_lookup(struct map *map, uint64_t key, uintptr_t *val);
+
+bool
+map_lookup_ref(struct map *map, uint64_t key, uintptr_t **val);
```

Comment:
The `_ptr` APIs are clearly convenience wrappers.  I can't tell from the API what this would be used for, in part bc there's add/remove to go with it.